tools/scripts/print_failures.py

```diff
@@ -9,29 +9,29 @@
 
-if len(sys.argv) != 2:
-    sys.exit(f"Usage: python {sys.argv[0]} [TEST_LOG_FILENAME|TEST_RESULTS.xml]")
 
-filename = sys.argv[1]
-extension = os.path.splitext(filename)[1]
-
-with open(filename) as file:
-    if extension.casefold() == ".xml".casefold():
-        test_xml = xml.dom.minidom.parse(file)
-        for testcase_elem in test_xml.getElementsByTagName("testcase"):
-            for failure_elem in testcase_elem.getElementsByTagName("failure"):
-                print("name: {}".format(testcase_elem.getAttribute("classname")))
-                print("output: {}".format(failure_elem.firstChild.data))
-                print("==================================================")
-    else:
-        test_log = json.load(file)
-        for result in test_log["tests"]:
-            if not result["code"] in ["PASS", "UNSUPPORTED", "XFAIL"]:
-                print("code: {}".format(result["code"]))
-                # Ignore result["elapsed"].
-                print("name: {}".format(result["name"]))
-                # The JSON contains embedded CRLFs (which aren't affected by opening the file in text mode).
-                # If we don't replace these CRLFs with LFs here, this script will appear to be okay in the console,
-                # but redirecting it to a file will result in ugly double newlines.
-                print("output: {}".format(result["output"].replace("\r\n", "\n")))
-                print("==================================================")
+if __name__ == "__main__":
+    if len(sys.argv) != 2:
+        sys.exit(f"Usage: python {sys.argv[0]} [TEST_LOG_FILENAME|TEST_RESULTS.xml]")
+    filename = sys.argv[1]
+    extension = os.path.splitext(filename)[1]
+    with open(filename) as file:
+        if extension.casefold() == ".xml".casefold():
+            test_xml = xml.dom.minidom.parse(file)
+            for testcase_elem in test_xml.getElementsByTagName("testcase"):
+                for failure_elem in testcase_elem.getElementsByTagName("failure"):
+                    print(f"name: {testcase_elem.getAttribute('classname')}")
+                    print(f"output: {failure_elem.firstChild.data}")
+                    print("==================================================")
+        else:
+            test_log = json.load(file)
+            for result in test_log["tests"]:
+                if not result["code"] in ["PASS", "UNSUPPORTED", "XFAIL"]:
+                    print(f"code: {result['code']}")
+                    # Ignore result["elapsed"].
+                    print(f"name: {result['name']}")
+                    # The JSON contains embedded CRLFs (which aren't affected by opening the file in text mode).
+                    # If we don't replace these CRLFs with LFs here, this script will appear to be okay in the console,
+                    # but redirecting it to a file will result in ugly double newlines.
+                    print(f"output: {result['output'].replace('\r\n', '\n')}")

```

Comment:
This breaks the script. How to repro:

1. Intentionally damage any STL test. Example:

```
C:\GitHub\STL\out\build\x64>git diff -U0
diff --git a/tests/std/tests/Dev11_0272959_make_signed/test.compile.pass.cpp b/tests/std/tests/Dev11_0272959_make_signed/test.compile.pass.cpp
index eff551f9..71a62cab 100644
--- a/tests/std/tests/Dev11_0272959_make_signed/test.compile.pass.cpp
+++ b/tests/std/tests/Dev11_0272959_make_signed/test.compile.pass.cpp
@@ -212 +212 @@ void example() {
-    STATIC_ASSERT(is_assignable_v<int&, short>);
+    STATIC_ASSERT(!is_assignable_v<int&, short>);
```

2. After building the STL, run that test by itself, and save a log file:

```
C:\GitHub\STL\out\build\x64>python tests\utils\stl-lit\stl-lit.py C:\GitHub\STL\tests\std\tests\Dev11_0272959_make_signed -o testing_x64.log
[...]
Testing Time: 3.73s
  Unsupported:  4
  Failed     : 27
```

3. Ask `print_failures.py` to print the log in a human-readable way. This now emits a `SyntaxError`:

```
C:\GitHub\STL\out\build\x64>python C:\GitHub\STL\tools\scripts\print_failures.py testing_x64.log
  File "C:\GitHub\STL\tools\scripts\print_failures.py", line 36
    print(f"output: {result['output'].replace('\r\n', '\n')}")
                                                             ^
SyntaxError: f-string expression part cannot include a backslash
```

For your convenience, I have uploaded this log file: [testing_x64.log](https://github.com/microsoft/STL/files/9642057/testing_x64.log)

Reply:
Syntax error now resolved; line change reverted.